thundra/wrappers/django/middleware.py

```diff
@@ -0,0 +1,45 @@
+from thundra import constants
+from thundra.context.execution_context_manager import ExecutionContextManager
+from thundra.plugins.invocation import invocation_support
+from thundra.wrappers.django.django_wrapper import DjangoWrapper, logger
+
+
+class ThundraMiddleware:
+    def __init__(self, get_response=None):
+        super().__init__()
+        self.get_response = get_response
+        self._wrapper = DjangoWrapper(disable_log=False)
+
+    def __call__(self, request):
+        setattr(request, '_thundra_wrapped', True)
+        # Code to be executed for each request before
+        # the view (and later middleware) are called.
+        before_done = False
+        try:
+            self._wrapper.before_request(request)
+            before_done = True
+        except Exception as e:
+            logger.error("Error during the before part of Thundra: {}".format(e))
+
+        response = self.get_response(request)
```

Comment:
might `get_response` fail?

Reply:
if so, should we call `after_request` and throw error here?

Reply:
By the documentation, it should not fail https://docs.djangoproject.com/en/3.1/topics/http/middleware/#exception-handling